core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -1203,46 +1200,42 @@ public Reader(CachableBlockFile.Reader rdr) throws IOException {
 
       lgContext = new LocalityGroupContext(currentReaders);
 
-      createHeap(currentReaders.length);
+      createHeap(currentReaders.size());
     }
 
-    private Reader(Reader r, LocalityGroupReader[] sampleReaders) {
-      super(sampleReaders.length);
+    private Reader(Reader r, List<LocalityGroupReader> sampleReaders) {
+      super(sampleReaders.size());
       this.reader = r.reader;
-      this.currentReaders = new LocalityGroupReader[sampleReaders.length];
+      this.currentReaders = new ArrayList<>(sampleReaders.size());
       this.deepCopies = r.deepCopies;
       this.deepCopy = false;
       this.readers = r.readers;
       this.sampleReaders = r.sampleReaders;
       this.samplerConfig = r.samplerConfig;
       this.rfileVersion = r.rfileVersion;
-      for (int i = 0; i < sampleReaders.length; i++) {
-        this.currentReaders[i] = sampleReaders[i];
-        this.currentReaders[i].setInterruptFlag(r.interruptFlag);
+      for (LocalityGroupReader reader : sampleReaders) {
+        reader.setInterruptFlag(r.interruptFlag);
+        this.currentReaders.add(reader);
       }
       this.lgContext = new LocalityGroupContext(currentReaders);
     }
 
     private Reader(Reader r, boolean useSample) {
-      super(r.currentReaders.length);
+      super(r.currentReaders.size());
       this.reader = r.reader;
-      this.currentReaders = new LocalityGroupReader[r.currentReaders.length];
+      this.currentReaders = new ArrayList<>(r.currentReaders.size());
       this.deepCopies = r.deepCopies;
       this.deepCopy = true;
       this.samplerConfig = r.samplerConfig;
       this.rfileVersion = r.rfileVersion;
       this.readers = r.readers;
       this.sampleReaders = r.sampleReaders;
 
-      for (int i = 0; i < r.readers.length; i++) {
-        if (useSample) {
-          this.currentReaders[i] = new LocalityGroupReader(r.sampleReaders[i]);
-          this.currentReaders[i].setInterruptFlag(r.interruptFlag);
-        } else {
-          this.currentReaders[i] = new LocalityGroupReader(r.readers[i]);
-          this.currentReaders[i].setInterruptFlag(r.interruptFlag);
-        }
-
+      List<LocalityGroupReader> readers = (useSample) ? r.sampleReaders : r.readers;
+      for (LocalityGroupReader reader : readers) {
+        LocalityGroupReader newReader = new LocalityGroupReader(reader);
```

Comment:
Good opportunity to use `var` if one were inclined.

```suggestion
        var newReader = new LocalityGroupReader(reader);
```